test/src/test/java/hudson/util/AtomicFileWriterPerfTest.java

```diff
@@ -0,0 +1,34 @@
+package hudson.util;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class AtomicFileWriterPerfTest {
+
+    @ClassRule
+    public static final JenkinsRule rule = new JenkinsRule();
+
+    /**
+     * This test is meant to catch huge regressions in terms of serialization performance.
+     * <p>
+     * <p>Some data points to explain the timeout value below:</p>
+     * <ul>
+     * <li>On a modern SSD in 2017, it takes less than a second to run.</li>
+     * <li>Using Docker resource constraints, and setting in read&write to IOPS=40 and BPS=10m (i.e. roughly worse than
+     * an old 5400 RPM hard disk), this test takes 25 seconds</li>
+     * </ul>
```

Comment:
FTR, takes roughly 1 second too in ci.jenkins.io:

![image](https://user-images.githubusercontent.com/223853/33368503-31b5f1ee-d4f2-11e7-879a-e771d987ba8a.png)